spec/models/user_spec.rb

```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it 'is valid with a name and a non-negative posts_counter' do
+    user = User.new(name: 'Tom', post_counter: 0)
+    expect(user).to be_valid
+  end
+
+  it 'is not valid without a name' do
+    user = User.new(name: nil)
+    user.valid?
+    expect(user.errors[:name]).to include("can't be blank")
+  end
+
+  it 'is not valid with a negative posts_counter' do
+    user = User.new(name: 'Tom', post_counter: -1)
+    user.valid?
+    expect(user.errors[:post_counter]).to include('must be greater than or equal to 0')
+  end
+
```

Comment:
---

Hello,

Great work on writing comprehensive tests for the User model! Here are some insights based on the project requirements and the code you've provided:

✅ **Positive Testing:**
- You've correctly tested that a user with a name and a non-negative `posts_counter` is valid. This aligns well with the validation requirements.

✅ **Negative Testing:**
- You've included tests to verify that a user without a name is not valid, as well as a test to ensure that the `posts_counter` cannot be negative. These are crucial negative test cases that properly validate model constraints.

🛠️ **Recommendations:**
- Include a test case to check if `posts_counter` is an integer. You can write a test that assigns a float or string to `posts_counter` and expects it not to be valid.

- Make sure that you have the actual validations in your User model file (`user.rb`). The tests will pass only if the validations are present in the model.

```ruby
# app/models/user.rb
class User < ApplicationRecord
  validates :name, presence: true
  validates :posts_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
end
```

- Don't forget to test other scenarios if required by the project specifications, like validations for other attributes, associations, or class and instance methods.

Keep up the good work! It's evident that you're applying TDD principles effectively, which is a key skill in software development.

---